go-controller/pkg/ovn/controller/apbroute/external_controller.go

```diff
@@ -47,6 +47,42 @@ type ExternalRouteInfo struct {
 	PodExternalRoutes map[string]map[string]string
 }
 
+type ExternalGatewayRouteInfoCache struct {
+	// External gateway caches
+	// Make them public so that they can be used by the annotation logic to lock on namespaces and share the same external route information
+	ExternalGWCache map[ktypes.NamespacedName]*RouteInfo
+	ExGWCacheMutex  *sync.RWMutex
+	routeInfoLocks  *syncmap.SyncMap[string]
```

Comment:
what is the relationship between routeInfoLocks and ExGWCacheMutex? In some cases I see ExternalGWCache controlled by one, and then in other instances controlled by the other.